File: C#/relatorio.cs
using System;
using System.IO;
using System.Net;
using System.Text;

class Program
{
    static void Main()
    {
        // URL do endpoint da API
        var url = "https://api.mailgrid.net.br/report/";

        try
        {
            // Criação da requisição HTTP POST
            var httpRequest = (HttpWebRequest)WebRequest.Create(url);
            httpRequest.Method = "POST";

            // Definindo os cabeçalhos da requisição
            httpRequest.Headers["Authorization"] = "Bearer SEU_TOKEN_AQUI"; // Cabeçalho de Autorização (caso necessário)
            httpRequest.ContentType = "application/json"; // Especificando que o conteúdo é JSON

            // Dados da requisição (parâmetros de consulta para o relatório)
            var data = "{\"usuario_smtp\": \"USUARIO-SMTP\", \"senha_smtp\": \"SENHA-SMTP\", " +
                       "\"dataini\": \"2022-09-27\", \"horaini\": \"00:01\", \"datafim\": \"2022-09-27\", \"horafim\": \"23:59\"}";

            // Escrevendo os dados no corpo da requisição
            using (var streamWriter = new StreamWriter(httpRequest.GetRequestStream(), Encoding.UTF8))
            {
                streamWriter.Write(data); // Envia os dados JSON como corpo da requisição
            }

            // Obtenção da resposta HTTP
            using (var httpResponse = (HttpWebResponse)httpRequest.GetResponse())
            {
                // Verificando se a resposta HTTP foi bem-sucedida
                if (httpResponse.StatusCode == HttpStatusCode.OK)
                {
                    // Lê o conteúdo da resposta
                    using (var streamReader = new StreamReader(httpResponse.GetResponseStream()))
                    {
                        var result = streamReader.ReadToEnd();
                        Console.WriteLine("Resposta da API:\n" + result); // Exibe a resposta da API
                    }
                }
                else
                {
                    Console.WriteLine($"Erro na requisição. Código de status HTTP: {httpResponse.StatusCode}");
                }
            }
        }
        catch (WebException webEx)
        {
            // Captura falhas de requisição, como erros de rede ou de resposta HTTP
            using (var streamReader = new StreamReader(webEx.Response.GetResponseStream()))
            {
                var errorResponse = streamReader.ReadToEnd();
                Console.WriteLine($"Erro na requisição: {errorResponse}");
            }
        }
        catch (Exception ex)
        {
            // Captura outros tipos de exceções, como problemas gerais
            Console.WriteLine($"Erro inesperado: {ex.Message}");
        }
    }
}

File: python/relatorio.py
import requests
from requests.structures import CaseInsensitiveDict

# URL do endpoint da API mailgrid para consultar relatórios
url = "https://api.mailgrid.net.br/report/"

# Criando um dicionário para os cabeçalhos HTTP da requisição
headers = CaseInsensitiveDict()
headers["Content-Type"] = "application/json"  # Cabeçalho que define o tipo de conteúdo como JSON

# Dados para a requisição POST - Contém informações sobre o período de consulta do relatório
data = '{"token_auth": "INFORME-O-TOKEN-AQUI, "dataini": "2022-09-27", "horaini": "00:01", "datafim": "2022-09-27", 
"horafim": "23:59"}'

# Fazendo a requisição POST para o endpoint com os cabeçalhos e os dados JSON
resp = requests.post(url, headers=headers, data=data)

# Verificando o código de status da resposta e exibindo o código de status HTTP
print(resp.status_code)

# Se a requisição for bem-sucedida (código 200), pode-se processar a resposta
if resp.status_code == 200:
	print("Requisição bem-sucedida!")  # Caso a resposta tenha sido bem-sucedida
	print(resp.json())  # Exibe a resposta JSON da API, que pode conter o relatório solicitado
else:
	# Caso ocorra um erro, exibe o erro com base no código de status
	print(f"Erro na requisição: {resp.text}")		
